core/sdk/src/api/device/entertainment_data.rs

```diff
@@ -662,9 +662,48 @@ pub enum ProgramEntityIntentData {
     Extra(AdditionalEntity),
 }
 
+// PlayableProgramEntity is almost similar to ProgramEntityIntentData.
+// firebolt-open-rpc.json has used a subset of ProgramEntity as PlayableEntity
+// Need to check if we can merge them
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+#[serde(tag = "programType", rename_all = "camelCase")]
+pub enum PlayableProgramEntity {
+    #[serde(alias = "Movie", alias = "MOVIE")]
+    Movie(MovieEntity),
+    #[serde(alias = "Episode", alias = "EPISODE")]
+    Episode(TVEpisodeEntity),
+    #[serde(alias = "Concert", alias = "CONCERT")]
+    Concert(AdditionalEntity),
+    #[serde(alias = "SportingEvent", alias = "SPORTINGEVENT")]
+    SportingEvent(AdditionalEntity),
+    #[serde(alias = "Preview", alias = "PREVIEW")]
+    Preview(AdditionalEntity),
+    #[serde(alias = "Other", alias = "OTHER")]
+    Other(AdditionalEntity),
+    #[serde(alias = "Advertisement", alias = "ADVERTISEMENT")]
+    Advertisement(AdditionalEntity),
+    #[serde(alias = "MusicVideo", alias = "MUSICVIDEO")]
+    MusicVideo(AdditionalEntity),
+    #[serde(alias = "Minisode", alias = "MINISODE")]
+    Minisode(AdditionalEntity),
+    #[serde(alias = "Extra", alias = "EXTRA")]
+    Extra(AdditionalEntity),
+}
+
 #[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct BaseEntity {
+    pub entity_type: String,
+    pub entity_id: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub asset_id: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub app_content_data: Option<AppContentDataString>,
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ProgramBaseEntity {
```

Comment:
Just curious why you need to define both `BaseEntity` and `ProgramBaseEntity` since they seem to be the same thing (except that `entity_type` of the former is a `String` and the latter is a `ProgramEntityType`, but that's really just a string with a default value).

Reply:
The default value for `entity_type` to "program" should be set only for a subset of entities which I categorized as `ProgramBaseEntity`.   The earlier implementation (BaseEntity) used to set the default entity_type value of all entities to `program`.  For example `playlist` is another valid entity_type.